### PR TITLE
ISPN-8238 Add spatial support to Ickle language

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-8238

~~Based on https://github.com/infinispan/infinispan/pull/13080, that should be merged first.~~ done!
~~Based on https://github.com/infinispan/infinispan/pull/13079, that should be merged first.~~ done!
~~Based on https://github.com/infinispan/infinispan/pull/13156, that should be merged first.~~ done!

~~Draft pull request. I want to open this PR in which I basically adapted all the work that Adrian did for this task.
There other stuff to do, and I'll add commits here:~~ done!

~~1. Support 100% remote spatial queries~~ done!
~~2. Support our annotations (for both embedded and remote queries)~~ done!
~~3. Support `@Point` binding (the alternative to `@Longitude` and `@Latitude`)~~ done!
~~4. Support multiple `@GeoPoint` with different markers for remote queries~~ done!
~~5. Support Box predicates (as alternative to `within round`)~~ done!
~~6. Support Polygon predicates (as alternative to `within round`)~~ done!
~~7. Test distance projections (can be very useful)~~ done!
~~8. Test sorting by distance (can be very useful)~~ done!
~~9. Document all the stuff (an issue in not completed without documenting it)~~ done!

### CODE REVIEW

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/indexing/search5/Search5Annotations.java [hunk 1]
```diff
@@ -31,6 +31,36 @@ public final class Search5Annotations {
    public static final String STORE_YES = "Store.YES";
    public static final String STORE_NO = "Store.NO";
 
+   public static final String PROJECTABLE_DEFAULT = "Projectable.DEFAULT";  // translates to either YES or NO depending on the indexing backend
+   public static final String PROJECTABLE_NO = "Projectable.NO";
```
Comment (tristantarrant) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/indexing/search5/Search5Annotations.java:35]: Do we really need to add stuff to the HS 5 compatibility layer ?

Comment (fax4ever) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/indexing/search5/Search5Annotations.java:35]: These rows will be overwritten by further commit messages, when we support the new annotations (point 2).

**Comment by fax4ever** [comment]
I added 3 substitutions for native compilations.

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/indexing/search5/Search5Annotations.java (hunk 1 above)
Comment (fax4ever) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/indexing/search5/Search5Annotations.java]: I removed with the last push. Basically I used this legacy code to have a working solution to replace with the proper final one.

**Comment by fax4ever** [comment]
rebased

**Comment by fax4ever** [comment]
green build

**Review by tristantarrant — APPROVED** [review]
I have made some changes to the docs.
It would be great to also prepare a companion PR for https://github.com/infinispan/infinispan-simple-tutorials which demos the feature.

**Comment by fax4ever** [comment]
> I have made some changes to the docs. It would be great to also prepare a companion PR for https://github.com/infinispan/infinispan-simple-tutorials which demos the feature.

Thanks @tristantarrant for the improvements here.

> It would be great to also prepare a companion PR for https://github.com/infinispan/infinispan-simple-tutorials which demos the feature.

Sure I can work on it.

**Comment by fax4ever** [comment]
Hi @Crumby I don't know if I can a feedback from you about the tests we added covering the spatial queries.

**Review by andyuk1986 — APPROVED** [review]
Hey @fax4ever , the tests look good to me, even though I would add more combinations of projections and sorting operations to queries. 
I have tried the functionality also on live server, tried more combinations for projections and sorting, tried with Hotrod Client and runnin queries over Console. Everything looks good. 
Thanks for your work.
Anna.

**Comment by fax4ever** [comment]
> Hey @fax4ever , the tests look good to me, even though I would add more combinations of projections and sorting operations to queries. I have tried the functionality also on live server, tried more combinations for projections and sorting, tried with Hotrod Client and runnin queries over Console. Everything looks good. Thanks for your work. Anna.

Thanks @andyuk1986 for the feedaback! Having this feedback at this stage i great or the process, I think.
Sure I can add more add more combinations of projections and sorting operations to queries.

**Comment by fax4ever** [comment]
I added some test trying in the same queries, spatial predicates + spatial projections + spatial sorting,
targeting different spatial field, for `@GeoField` and `@GeoPoint`.
I don't know if we should other test cases. Thanks again @andyuk1986

**Comment by fax4ever** [comment]
Thanks for all the reviews and for merging it!

**Pull request merged**